src/ore/src/task.rs

```diff
@@ -0,0 +1,174 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License in the LICENSE file at the
+// root of this repository, or online at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::future::Future;
+use std::sync::Arc;
+
+use tokio::runtime::{Handle, Runtime};
+use tokio::task::{self, JoinHandle};
+
+/// Spawns a task on the executor.
+///
+/// See [`spawn`](::tokio::spawn) for
+/// more details.
+#[cfg(not(all(tokio_unstable, feature = "task")))]
+#[track_caller]
+pub fn spawn<Fut, Name, NameClosure>(_nc: NameClosure, future: Fut) -> JoinHandle<Fut::Output>
+where
+    Name: AsRef<str>,
+    NameClosure: FnOnce() -> Name,
+    Fut: Future + Send + 'static,
+    Fut::Output: Send + 'static,
+{
+    tokio::spawn(future)
+}
+
+/// Spawns a task on the executor.
+///
+/// See [`spawn`](tokio::spawn) for
+/// more details.
+#[cfg(all(tokio_unstable, feature = "task"))]
+#[track_caller]
+pub fn spawn<Fut, Name, NameClosure>(nc: NameClosure, future: Fut) -> JoinHandle<Fut::Output>
```

Comment:
Maybe I missed it, but what actually requires `nc` to be a closure? It seems like in all cases it could be just an `AsRef<str>`?

Reply:
I am just trying to avoid as many allocations as possible, as many names use `format!` and the name won't be used in non-`--tokio-console` cases

Reply:
Ahh I see. Makes sense. Carry on!